### ISSUE TITLE
kwallet is not unlocked with SDDM

### ISSUE THREAD

**Issue by xfix** [NixOS/nixpkgs#101904]
**Describe the bug**

After logging in with SDDM I see the following window:

![](https://user-images.githubusercontent.com/8165792/77233680-54870000-6ba9-11ea-8349-1593efb44d48.png)

Every time. This is annoying and should not happen. Previously, no manual intervention was necessary. This is a regression in 20.09.

This is like #83095, but for 20.09 this time.

**To Reproduce**
1. Enable NetworkManager, SDDM and Plasma.
2. Store Wi-Fi password in KDE Wallet (store password for this user only - encrypted).
3. Set KDE Wallet password to password used to log in.
4. Log out.
5. Observe the above-mentioned window.

**Expected behavior**

No manual intervention should be necessary.

**Notify maintainers**

@ttuegel @nyanloutre 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.72, NixOS, 20.09.1469.13d0c311e3a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(xfix): `""`
 - channels(root): `"nixos-20.09.1469.13d0c311e3a, home-manager-20.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: plasma5
```

**Comment by worldofpeace** [comment]
Yeah, I think Q.A finding showed this was true.

**Comment by worldofpeace** [comment]
cc @ttuegel

**Comment by doronbehar** [comment]
It's possible this may be fixed with https://github.com/NixOS/nixpkgs/pull/101369 (backport https://github.com/NixOS/nixpkgs/pull/102347 ) because IIRC before the changes there sddm didn't  the same qt version as the other plasma packages... But we need to check it..

**Comment by matthewbauer** [comment]
Ugh.

**Comment by matthewbauer** [comment]
Don't have much time to look at this, but last time a few things pieces of info that could help:

- Check for kwallet-pam in coredumpctl (the coredump itself won't be very helpful, but at least we know it's getting that far.
- Try increasing the kwallet-pam max from 1000 to 10000 in https://github.com/KDE/kwallet/blob/bc9713e2725ab1c4311866f751c674a38584bd92/src/runtime/kwalletd/main.cpp#L44
- Might want check if any env vars are over the 1000 char limit. They get cut off which can lead to kwallet-pam crash.

**Comment by ttuegel** [comment]
> It's possible this may be fixed with #101369 (backport #102347 ) because IIRC before the changes there sddm didn't the same qt version as the other plasma packages...

They didn't use different Qt versions before the branch-off, but this has been broken much much longer than that.

**Comment by xfix** [comment]
I think 21.05 fixes this, so closing.
